pandas/core/dtypes/cast.py

```diff
@@ -996,7 +1016,9 @@ def astype_nansafe(arr, dtype, copy: bool = True, skipna: bool = False):
     return arr.view(dtype)
 
 
-def maybe_convert_objects(values: np.ndarray, convert_numeric: bool = True):
+def maybe_convert_objects(
+    values: np.ndarray, convert_numeric: bool = True
+) -> Union[np.ndarray, ABCDatetimeIndex]:
```

Comment:
use "DatetimeIndex" instead of ABCDatetimeIndex

Reply:
Done